src/main.c

```diff
@@ -297,7 +297,7 @@ run(const struct command *command, int argc, char **array)
 	if (mpd_connection_cmp_server_version(conn, 0, 21, 0) < 0)
 		fprintf(stderr, "warning: MPD 0.21 required\n");
 
-	if (options.partition != NULL) {
+	if (options.partition != NULL && command->handler != cmd_moveoutput) {
```

Comment:
This is a very obscure check (comparing function pointers is always obscure and questionable, but for me an acceptable kludge here) and deserves a code comment.

Reply:
Added.